File: C++/MEDIUM/1545. Find Kth Bit in Nth Binary String.cpp
class Solution
{
public:
    char findKthBit(int n, int k)
    {
        int indicator;
        int middle = pow(2, n - 1);
        if (n == 1)
        {
            indicator = 0;
        }
        else
        {
            if (k == middle)
            {
                indicator = 1;
            }
            else if (k > middle)
            {
                indicator = findKthBit(n - 1, middle - (k - middle)) == '0';
            }
            else if (k < middle)
            {
                indicator = findKthBit(n - 1, k) == '1';
            }
        }
        return '0' + indicator;
    }
};

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.java
class Solution {
    private int gets(int n, int k){
        if (n == 1) return 0;
        int l = (1 << n) - 1;
        if (k == l / 2)
        {
            return 1;
        }
        else if (k < l / 2)
        {
            return gets(n - 1, k);
        }
        else
        {
            return (
            // 여기서 한번 뒤집어줘야하는구나 
            gets(n - 1, 
            ((l >> 1) - 1) - (k - (l / 2 + 1))) + 1) % 2;
        }
    }
    public char findKthBit(int n, int k) {
        return(char) ('0' + gets(n, k - 1));
    }
}